BlockchainSdk/Common/Interfaces/TransactionValidator/TransactionValidator.swift

```diff
@@ -169,3 +169,12 @@ extension TransactionValidator where Self: ReserveAmountRestrictable {
         }
     }
 }
+
+// MARK: - FeeResourceRestrictable
+
+extension TransactionValidator where Self: FeeResourceRestrictable {
+    func validate(amount: Amount, fee: Fee) async throws {
+        try validateAmounts(amount: amount, fee: fee.amount)
+        try await validateFeeResource(amount: amount, fee: fee.amount)
+    }
+}
```

Comment:
Тут надо скопировать с `DustRestrictable`. Так как не везде еще используется async валиация.

там два метода, оба обязательны

```suggestion
extension TransactionValidator where Self: FeeResourceRestrictable {
    func validate(amount: Amount, fee: Fee, destination: DestinationType) async throws {
        Log.debug("TransactionValidator \(self) doesn't checking destination. If you want it, make our own implementation")
        try validate(amount: amount, fee: fee)
    }
    
    func validate(amount: Amount, fee: Fee) throws {
        try validateAmounts(amount: amount, fee: fee.amount)
        try validateFeeResource(amount: amount, fee: fee.amount)
    }
}
```